### PR TITLE
File extensions

### PR DESCRIPTION
**Pull request by jonatanklosko**
Improves the *Open in separate editor* feature.

This adds an ability of inferring the appropriate file extension basing on the response Content-Type. The most important point is that this allows Atom to highlight the file nicely and makes working with it easier. (Quiet related with #37.)

I also moved the option to the top as it's really frustrating to scroll down when the response is huge. Here's how it looks now:

screenshot from 2016-10-10 19-39-59

screenshot from 2016-10-10 19-40-12

### CODE REVIEW

**Comment by ddavison** [comment]
@javaguirre @bernardlee ?

File: lib/rest-client-view.coffee [hunk 1]
```diff
@@ -194,10 +196,13 @@ class RestClientView extends ScrollView
 
   openInEditor: ->
     textResult = $(rest_form.result).text()
-    file_name = "#{current_method} - #{$(rest_form.url).val()}"
-    editor = new RestClientEditor(textResult, file_name)
+    editor = new RestClientEditor(textResult, @constructFileName())
     editor.open()
 
+  constructFileName: ->
+    extension = mime.extension(@lastResponse.headers['content-type']);
```
Comment (javaguirre) [inline lib/rest-client-view.coffee]: This `;` is not needed

File: lib/rest-client-view.coffee [hunk 2]
```diff
@@ -194,10 +196,13 @@ class RestClientView extends ScrollView
...
+    "#{@lastRequest.method} #{@lastRequest.url}#{'.' + extension if extension}"
```
Comment (javaguirre) [inline lib/rest-client-view.coffee]: This is kind of unreadable, could you rewrite it in a ways It's easier to read? Than you! :-)

Comment (javaguirre) [inline lib/rest-client-view.coffee]: Nevermind, I think It's fine this way 👍

**Pull request merged**